ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala

```diff
@@ -469,7 +470,49 @@ private[testtool] abstract class CommandDeduplicationBase(
 
 object CommandDeduplicationBase {
   trait DelayMechanism {
-    def apply(): Future[Unit]
+    val deduplicationDuration: Duration
+    val extraWait: Duration
+
+    /** Delay by the guaranteed full deduplication period. After calling this method any duplicate calls should succeed
+      */
+    def delayForEntireDeduplicationPeriod(): Future[Unit] =
+      delayBy(deduplicationDuration + extraWait)
+
+    /** Delay by [[deduplicationDuration]] - [[delta]]. Delta should be a value big enough to account for the latency required to process a transaction.
+      * By using this delay we can validate that the minimum deduplication period [[deduplicationDuration]] is respected
+      */
+    def delayToEndOfMinimumDeduplicationPeriod(delta: Duration): Future[Unit] =
```

Comment:
Where do we need these `delayTo*` and `delayFrom*`  methods?

Reply:
Not currently used, I had previously added a test case that was using them but decided to leave that test case for a separate different PR where we would use these methods as well.

Reply:
Can we introduce these methods in the PR which is going to use them then, please? This should help in reducing the scope of this PR as well.